packages/iov-bns/src/encode.ts

```diff
@@ -513,7 +515,7 @@ export function buildUnsignedTx(tx: UnsignedTransaction): codecImpl.bnsd.ITx {
       tx.fee && tx.fee.tokens
         ? {
             fees: encodeAmount(tx.fee.tokens),
-            payer: feePayer,
+            payer: feePayerAddressBytes,
           }
```

Comment:
What about moving the payer into the `Fee` object? It ensures no payer is set if no fee is set. Here you see that this would be closer to the weave implementation.